llvm/docs/HowToCrossCompileLLVM.rst

```diff
@@ -6,210 +6,202 @@ Introduction
+   ``install`` target. But setting it avoids installing binaries compiled for
+   the target across system directories on the host system.
+
+See `LLVM's build documentation
+<https://llvm.org/docs/CMake.html#frequently-used-cmake-variables>`_ for more
+guidance on CMake variables (e.g. ``LLVM_TARGETS_TO_BUILD`` may be useful if
+your cross-compiled binaries only need to support compiling for one target).
+
+Assuming you have ``qemu-user`` installed you can test the produced target
+binaries either by relying on binfmt_misc (as was necessary for debootstrap)
+or invoking ``qemu-{tgt}-static`` directly. For instance, to first check the
+type of executable and then run ``clang --version`` both ways:
 
    .. code-block:: bash
 
-     $ ninja
+    $ file -L ./build/aarch64-linux-gnu/bin/clang
+    ./build/aarch64-linux-gnu/bin/clang: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, BuildID[sha1]=516b8b366a790fcd3563bee4aec0cdfcb90bb1c7, not stripped
+
+    $ qemu-aarch64-static -L $SYSROOT ./build/aarch64-linux-gnu/bin/clang --version
+    clang version 21.0.0git (https://github.com/llvm/llvm-project cedfdc6e889c5c614a953ed1f44bcb45a405f8da)
+    Target: aarch64-unknown-linux-gnu
+    Thread model: posix
+    InstalledDir: /home/asb/llvm-project/build/aarch64-linux-gnu/bin
+
+    $ export QEMU_LD_PREFIX=$SYSROOT; ./build/aarch64-linux-gnu/bin/clang --version
+    clang version 21.0.0git (https://github.com/llvm/llvm-project cedfdc6e889c5c614a953ed1f44bcb45a405f8da)
+    Target: aarch64-unknown-linux-gnu
+    Thread model: posix
+    InstalledDir: /home/asb/llvm-project/build/aarch64-linux-gnu/bin
+
+Working around a ninja dependency issue
+---------------------------------------
+
+If you followed the instructions above to create a sysroot, you may run into a
+`longstanding problem related to path canonicalization in ninja
+<https://github.com/ninja-build/ninja/issues/1330>_`. GCC canonicalizes system
+headers in dependency files, so when ninja reads them it doesn't need to do
+so. Clang doesn't, and unfortunately ninja doesn't implement the
+canonicalization logic at all, meaning for some system headers with symlinks
+in the paths, it can incorrectly compute a non-existing path and consider it
+as always dirty. If you are suffering from this issue, you will find at any
+attempt at an incremental build (including the suggested command to build the
+``install`` target in the next section) results in recompiling everything.
+``ninja -C build/$TARGET -t deps`` shows files in ``$SYSROOT/include/*`` that
+don't (as the ``$SYSROOT/include`` folder doesn't exist) and you can
+further confirm these files are causing the build to be seen as dirty with
+``ninja -C build/$TARGET -d deps``. Until such time as the issue is avoided
+due to a change in behaviour for Clang or for Ninja, a functional workaround
```

Comment:
I might just start with "A workaround is to..." and delete the first part.

If this sentence is separated from the rest it will clearly be the "and here's how to deal with it" bit.

Reply:
I've applied this suggested edit.